README.md

```diff
@@ -20,224 +20,235 @@
-                d.convert_uri_to_datauri()  # convert to datauri
-                d.pop('embedding', 'blob')  # remove unnecessary fields for save bandwidth
-    ```
-5. Connect all `Executor`s in a `Flow`, scale embedding to 3:
-    ```python
-    f = Flow(port_expose=12345, protocol='http').add(uses=PreprocImg).add(uses=EmbedImg, replicas=3).add(uses=MatchImg)
-    ```
-    Plot it via `f.plot('flow.svg')` and you get:
-    ![](.github/images/readme-flow-plot.svg)
+with Flow().add(uses=ImageEmbeddingExecutor).add(uses=IndexExecutor) as f:
+    left_da = DocumentArray.from_files('~/Downloads/left/*.jpg')
+    right_da = DocumentArray.from_files('~/Downloads/right/*.jpg')
+    client = Client(port=f.port_expose)
+    client.index(left_da)
+        
+    response = client.search(right_da[:1])
+    docs = response[0].docs
+    right_da[:1].plot_image_sprites()
+    (DocumentArray(docs[0].matches, copy=True)
+         .apply(lambda d: d.set_image_tensor_channel_axis(0, -1)
+                .set_image_tensor_inv_normalization())
+         .plot_image_sprites())
+```
+You will see the image you searched for and the top 10 matches. Just close the images as they show up. This is everything, you just build your first neural search application! 🎉
 
-6. Index image data and serve REST query publicly:
+4. You can also expose your application with a REST API so that you can send HTTP request: Just change the protocol of the `Flow` to HTTP and use Curl to query it:
+```python
+from jina import Client, Flow  
+   
+with Flow(port_expose=12345, protocol='http').add(uses=ImageEmbeddingExecutor).add(uses=IndexExecutor) as f:
+    left_da = DocumentArray.from_files('~/Downloads/left/*.jpg')
+    client = Client(port=f.port_expose, protocol='http')
+    client.index(left_da[:10])
+    f.block()
+```
+6. Now use Curl to send search requests:
+```bash
+curl -X POST http://127.0.0.1:12345/search -H 'Content-type: application/json' -d '{"data":[{"uri": "<data_set_path>/right/00000.jpg"}]}' > curl_response
+```
+
+### Containerize your Executors
+If we want to be able to use our Executors via Docker Compose or in K8s we will need to containerize them. The easiest way to do that is the usage of the [Jina Hub](https://hub.jina.ai).
+1. Move each `Executor` class to a separate folder with one Python file in each:
+   - `ImageEmbeddingExecutor` -> 📁 `embed_img/exec.py`
+   - `IndexExecutor` -> 📁 `match_img/exec.py`
+2. Create a `requirements.txt` in embed_img and add `torchvision` as requirement there.
+3. Push all Executors to [Jina Hub](https://hub.jina.ai). **Important**: Write down the string you get for the Usage. It looks like this `jinahub://1ylut0gf`:
+```bash
+jina hub push embed_img
+jina hub push match_img
+```
+   You will get two Hub Executors that can be used for any container.
+5. Now you are able run the `Flow` from the previous example with your containerized Executors. Just replace the `uses` strings with the values you got from Jina Hub:
```

Comment:
Maybe u can add the `command` to parse the output of `jina hub push` to keep the `jinahub` IDs inside `ENVIRONMENT` variables to be used later in the next script?